packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue

```diff
@@ -0,0 +1,359 @@
+            this.legacyBanner = this.config.displayLegacy;
+        },
+        /**
+         * Check if the cookie banner has been shown to this user
+         */
+        checkCookieBannerCookie () {
+            if (this.legacyBanner) {
+                this.shouldHideBanner = this.$cookies.get(this.legacyConsentCookieName) === 2;
+                this.setLegacyCookieBannerCookie();
+            } else {
+                const cookieConsent = this.$cookies.get(this.consentCookieName);
+                this.shouldHideBanner = cookieConsent === 'full' || cookieConsent === 'necessary';
+            }
+        },
+        /**
+         * Set the cookie for the user's choice
+         * @param {String} cookieValue
+         */
+        setCookieBannerCookie (cookieValue) {
+            this.$cookies.set(this.consentCookieName, cookieValue, {
+                path: '/',
+                maxAge: this.cookieExpiry
+            });
+        },
+        /**
+         * Set the legacy cookie banner cookie
+         */
+        setLegacyCookieBannerCookie () {
+            this.$cookies.set(legacyConsentCookieName, '2', {
+                path: '/',
+                maxAge: this.cookieExpiry
+            });
+        },
+        /**
+         * Push tracking events
+         * @param {String} consentLevel
+         */
+        dataLayerPush (consentLevel) {
+            const dataLayer = window.dataLayer || [];
+            dataLayer.push({ event: 'trackConsent', userData: { consent: consentLevel } });
+            dataLayer.push({ platformData: { consentLoading: true } });
+        },
+        /**
+         * Check whether a cookie with name `cookieName` is to be excluded from being removed
+         * @param {String} cookieName
+         * @returns {Bool}
+         */
+        isNotExcluded (cookieName) {
+            return this.config.cookieExclusionList.every(excludedCookieName => cookieName.lastIndexOf(excludedCookieName, excludedCookieName.length - 1) === -1);
+        },
+        /**
+         * Remove unnecessary cookies
+         */
+        removeUnnecessaryCookies () {
+            const cookies = Object.keys(this.$cookies.getAll());
+            for (let i = 0; i < cookies.length; i++) {
+                if (this.isNotExcluded(cookies[i])) this.$cookies.remove(cookies[i]);
+            }
+        },
+        /**
+         * Check for iOS browser
+         * @returns {Bool}
+         */
+        isIosBrowser () {
+            return /(iPhone|iPad).*Safari/.test(navigator.userAgent);
+        },
+        /**
+         * Resend GTM events
+         */
+        resendEvents () {
+            const dataLayer = window.dataLayer || [];
+            const excludeEvents = ['gtm.js', 'gtm.click', 'gtm.linkClick', 'gtm.triggerGroup', 'trackConsent', 'showConsent'];
+            let i;
+            for (i = 0; i < dataLayer.length; i++) {
+                if (dataLayer[i].event && !excludeEvents.includes(dataLayer[i].event)) {
```

Comment:
Can add `dataLayer[i]` to a variable and reference that rather than repeat the lookup throughout the loop.

Reply:
@DamianMullins This is third party code, which I have pasted in as supplied. I'll ask the GTM guys if they can change this, otherwise, we'll have to document what we change and why in case we need them to help troubleshoot at any time.